### PR TITLE
Ensure integer paths produce arrays

### PR DESCRIPTION
**Pull request by taras**
# Motivation

Ramda lenses allow you to create arrays by passing integers to `lensPath` lens constructor. This is a helpful way to construct context when providing a path. I also noticed that when transitioning an ArrayType Microstate, the result of the transition has an object in the value instead of an array. I believe this stems from the same problem.

# Approach

When creating context for a value, I detect if path is an integer and create an array context instead of an object.

### CODE REVIEW

File: tests/lens.test.js [hunk 1]
```diff
@@ -1,25 +1,155 @@
 /* global describe, it, beforeEach */
-import expect from 'expect';
-import { compose, view, set, At } from '../src/lens';
+import expect from "expect";
+import { compose, view, set, At, Path, over } from "../src/lens";
 
-describe('At', function() {
+describe("At", function() {
```
Comment (cowboyd) [inline tests/lens.test.js:5]: Let's make cosmetic fixes in a separate PR that updates our linting rules.

**Review by cowboyd — COMMENTED** [review]
This is why the `At` function has different behavior based on the `context`, not the `property`.

In other words, the current behavior allows you to do this already:

```js
let arrayLens = At(5, [])
let objectLens = At(5, {});

set(objectLens, 'five', undefined) //=> {'5': 'five'}
set(arrayLens, 'five', undefined) //=> [,,,,'five']
```
This is more powerful than Rambda lenses, because it lets you capture an entire _actual_ structure as a lens, and then have it create the right combination of objects and arrays from scratch, even if the objects have number properties and the arrays have string properties.

In fact, in a prior version, the `At` function that created a lens was [fully polymorphic][1], so the structure that a lens could capture was more than just `Object`s and `Array`s, but _any combination of JavaScript values_. Notice how `Object` and `Array` have [different implementations of the `At` type class][2].

We removed this polymorphism in order to reduce complexity, but we could introduce it back again.

[1]: https://github.com/microstates/microstates.js/blob/3d7af65cafab355c4893eb4c7ae0c7b991de0e7f/src/lens.js#L86-L90

[2]: https://github.com/microstates/microstates.js/blob/3d7af65cafab355c4893eb4c7ae0c7b991de0e7f/src/lens.js#L114-L132

**Comment by taras** [comment]
@cowboyd I'll close this PR and create another PR that only has added tests. Then create a separate PR for ESlint changes.
